test/e2e/apimachinery/flowcontrol.go

```diff
@@ -0,0 +1,129 @@
+	f := framework.NewDefaultFramework("flowschemas")
+
+	ginkgo.It("should ensure that requests can be classified by testing flow-schemas/priority-levels", func() {
+		testingFlowSchemaName := "e2e-testing-flowschema"
+		testingPriorityLevelName := "e2e-testing-prioritylevel"
+		matchingUsername := "noxu"
+		nonMatchingUsername := "foo"
+
+		ginkgo.By("creating a testing prioritylevel")
+		createdPriorityLevel, err := f.ClientSet.FlowcontrolV1alpha1().PriorityLevelConfigurations().Create(
+			context.TODO(),
+			&flowcontrolv1alpha1.PriorityLevelConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testingPriorityLevelName,
+				},
+				Spec: flowcontrolv1alpha1.PriorityLevelConfigurationSpec{
+					Type: flowcontrolv1alpha1.PriorityLevelEnablementLimited,
+					Limited: &flowcontrolv1alpha1.LimitedPriorityLevelConfiguration{
+						AssuredConcurrencyShares: 1, // will have at minimum 1 concurrency share
+						LimitResponse: flowcontrolv1alpha1.LimitResponse{
+							Type: flowcontrolv1alpha1.LimitResponseTypeReject,
+						},
+					},
+				},
+			},
+			metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("creating a testing flowschema")
+		createdFlowSchema, err := f.ClientSet.FlowcontrolV1alpha1().FlowSchemas().Create(
+			context.TODO(),
+			&flowcontrolv1alpha1.FlowSchema{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testingFlowSchemaName,
+				},
+				Spec: flowcontrolv1alpha1.FlowSchemaSpec{
+					MatchingPrecedence: 1000, // a rather higher precedence to ensure it make effect
+					PriorityLevelConfiguration: flowcontrolv1alpha1.PriorityLevelConfigurationReference{
+						Name: testingPriorityLevelName,
+					},
+					DistinguisherMethod: &flowcontrolv1alpha1.FlowDistinguisherMethod{
+						Type: flowcontrolv1alpha1.FlowDistinguisherMethodByUserType,
+					},
+					Rules: []flowcontrolv1alpha1.PolicyRulesWithSubjects{
+						{
+							Subjects: []flowcontrolv1alpha1.Subject{
+								{
+									Kind: flowcontrolv1alpha1.SubjectKindUser,
+									User: &flowcontrolv1alpha1.UserSubject{
+										Name: matchingUsername,
+									},
+								},
+							},
+							NonResourceRules: []flowcontrolv1alpha1.NonResourcePolicyRule{
+								{
+									Verbs:           []string{flowcontrolv1alpha1.VerbAll},
+									NonResourceURLs: []string{flowcontrolv1alpha1.NonResourceAll},
+								},
+							},
+						},
+					},
+				},
+			},
+			metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("response headers should contain flow-schema/priority-level uid")
+
+		if !testResponseHeaderMatches(f.ClientConfig(), matchingUsername, string(createdPriorityLevel.UID), string(createdFlowSchema.UID)) {
+			framework.Failf("matching user doesnt received UID for the testing priority-level and flow-schema")
+		}
+		if !testResponseHeaderMatches(f.ClientConfig(), nonMatchingUsername, string(createdPriorityLevel.UID), string(createdFlowSchema.UID)) {
```

Comment:
shouldn't this be `if testResponseHeaderMatches`?

Reply:
LGTM otherwise. Let's see the test result.

Reply:
fixed